module/VuFind/src/VuFind/AjaxHandler/GetRecordCover.php

```diff
@@ -102,11 +117,24 @@ public function handleRequest(Params $params)
             );
         }
 
-        return $this->formatResponse(
-            [
-                'url' => $this->coverRouter->getUrl($record, $size ?? 'small'),
-                'size' => $size,
-            ]
-        );
+        $url = $this->coverRouter->getUrl($record, $size ?? 'small',true,$this->useCoverFallbacksOnFail);
+
+        if ($url || !$this->useCoverFallbacksOnFail) {
```

Comment:
This `if..else` might be slightly more concise as something like:

<pre>
return ($url || !$this->useCoverFallbacksOnFail)
    ? $this->formatResponse( ... first options ... )
    : $this->formatResponse ( ... second options ... )
</pre>